core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java

```diff
@@ -332,7 +333,7 @@ else if (nonce.isPresent() != oauth2Response.getIdToken().isPresent()) {
         if (userinfoUri.isPresent()) {
             // validate access token is trusted by remote userinfo endpoint
             Request request = Request.builder()
-                    .setMethod(POST)
+                    .setMethod(userInfoRequestMethod)
```

Comment:
Can we simply change it to GET?

Reply:
Both methods are allowed according to the spec.

Reply:
According to https://openid.net/specs/openid-connect-core-1_0.html#UserInfoRequest GET is recommended. The thing that we can use both methods does not mean we need to. I would prefer to have less toggles and use GET if is more widely. The issue is that we need to make sure we don't introduce regression.